### PR TITLE
feat: add billing event to front payment

### LINKED ISSUE #6898: Fixing IDE red lines
### Describe the bug

PHPStorm was giving a red line under this file every time I opened it. So, here are the fixes.

### To Reproduce

1. Add a '....'

### Client configuration

- Browser:
- OpenEMR version:
- Operating system:

### PR DESCRIPTION
**Pull request by juggernautsei**
Fixes #6898

### CODE REVIEW

File: interface/patient_file/front_payment.php [hunk 1]
```diff
@@ -1276,9 +1281,9 @@ function make_insurance() {
                                         $encs[$key]['charges'] += $brow['fee'];
                                         // Add taxes.
                                         $sql_array = array();
-                                        $query = "SELECT taxrates FROM codes WHERE " .
+                                        $query = "SELECT taxrates FROM `codes` WHERE " .
                                         "code_type = ? AND " .
-                                        "code = ? AND ";
+                                        "code = ? AND " .
```
Comment (stephenwaite) [inline interface/patient_file/front_payment.php:1286]: hi @juggernautsei , this looks odd here.

Comment (juggernautsei) [reply interface/patient_file/front_payment.php:1286]: You are right but the IDE tags it with a red underline (Which the IDE can be wrong). @stephenwaite.
I will revert it back.

File: interface/patient_file/front_payment.php [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<?php
+<?php global $srcdir, $pid, $encounter, $cryptoGen;
```
Comment (bradymiller) [inline interface/patient_file/front_payment.php:1]: Above line seems odd. Why need to do this?

**Comment by bradymiller** [comment]
btw, wouldn't worry too much about IDE redline stuff. As long as passing PSR stuff (which is checked by PR ci testing), then should be good.

File: interface/patient_file/front_payment.php (hunk 2 above)
Comment (sjpadgett) [reply interface/patient_file/front_payment.php:1]: prob prevent IDE from complaining otherwise not good practice!

**Comment by juggernautsei** [comment]
There were some things that needed changing. I will roll back the changes from the IDE and submit only the things that need updating. IDE is PHPStorm which was promoted to be used.

File: interface/patient_file/front_payment.php (hunk 2 above)
Comment (juggernautsei) [reply interface/patient_file/front_payment.php:1]: https://github.com/openemr/openemr/pull/6899#:~:text=%3C%3Fphp%20global%20%24srcdir%2C%20%24pid%2C%20%24encounter%2C%20%24cryptoGen%3B
PHPStorm suggested this.

Comment (bradymiller) [reply interface/patient_file/front_payment.php:1]: Prob best to remove that redline feature in PHPStorm or to ignore it :)

File: interface/patient_file/front_payment.php [hunk 3]
```diff
@@ -29,6 +29,11 @@
 use OpenEMR\OeUI\OemrUI;
 use OpenEMR\PaymentProcessing\Sphere\SpherePayment;
 use OpenEMR\Services\FacilityService;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+/**
+ * @var EventDispatcher
+ */
+$ed = $GLOBALS['kernel']->getEventDispatcher();
```
Comment (bradymiller) [inline interface/patient_file/front_payment.php:36]: remove above changes and change line 1895 to:
```
$GLOBALS['kernel']->getEventDispatcher()->dispatch(new PostFrontPayment(), PostFrontPayment::ACTION_POST_FRONT_PAYMENT, 10);
```

**Comment by stephenwaite** [comment]
hi @juggernautsei , you could rename this PR to something like you did for https://github.com/openemr/openemr/commit/74366f37fe7601d50d7057871003e02a58b152ce

```
feat: add billing event to front payment
```

[changelog reference pr](https://github.com/openemr/openemr/issues/6141)

**Comment by stephenwaite** [comment]
happy_monkey

**Pull request merged**